### PR TITLE
[ENH] Raise more meaningful configuration warnings

### PR DESCRIPTION
**Pull request by richard-willdooit**

### CODE REVIEW

File: product_configurator/models/product.py [hunk 1]
```diff
@@ -369,19 +370,26 @@ def value_available(self, attr_val_id, value_ids):
...
-            else:
-                if set(domain[2]) & set(value_ids):
-                    return False
+        for config_line in config_lines:
+            domains = config_line.mapped('domain_id').compute_domain()
+            for domain in domains:
+                if domain[1] == 'in':
+                    if not set(domain[2]) & set(value_ids):
+                        # Don't break if the module has not been upgraded...
+                        if do_raise and config_line.rule_description:
```
Comment (PCatinean) [inline product_configurator/models/product.py]: I think you accidentally removed the fields check here

Comment (richard-willdooit) [reply product_configurator/models/product.py]: Ahh, not, not quite.  What I discovered is, that it is in fact redundant!  It fails in any case. Sure, upgrading the module puts the fields in the db, but restarting the server finds the field definitions and therefore puts them in the _fields list.  So in fact, running the code without upgrading the module still has an issue...

Comment (richard-willdooit) [reply product_configurator/models/product.py]: I forgot to take out the comment :blush:

**Comment by PCatinean** [comment]
@richard-willdooit I have merged @matt454357 PR on the domain improvement.

Can you rebase on his changes?

**Pull request closed without merging**